src/evo/deterministicmns.cpp

```diff
@@ -1667,50 +1667,48 @@ bool CheckProUpServTx(const CTransaction& tx, const CBlockIndex* pindexPrev, TxV
         }
     }
 
-    if (pindexPrev) {
```

Comment:
Strictly saying, this is a change in behaviour and not a refactor (same for other `CheckPro...Tx`s). But it doesn't crash here as far as I see so ok.

Reply:
Not really as the function would crash earlier.